python/ray/serve/controller.py

```diff
@@ -701,6 +705,14 @@ def get_serve_instance_details(self) -> Dict:
         # route_prefix is set instead in each application.
         # Eventually we want to remove route_prefix from DeploymentSchema.
         return ServeInstanceDetails(
+            controller_info=ServeActorDetails(
+                node_id=ray.get_runtime_context().get_node_id(),
+                node_ip=ray.util.get_node_ip_address(),
+                actor_id=ray.get_runtime_context().get_actor_id(),
+                actor_name=self.controller_name,
+                worker_id=ray._private.worker.global_worker.worker_id.hex(),
+                log_file_path=get_component_logger_file_path(),
+            ),
```

Comment:
these fields are all immutable, suggest to store this in the constructor to avoid reconstructing every time:
```python
self._actor_details = ServeActorDetails(...)
```